src/service/websocket/socket_manager.rs

```diff
@@ -81,7 +84,17 @@ impl<'a> SocketManager<'a> {
         }
     }
 
-    async fn blocking_listen(&mut self) -> Result<(), std::io::Error> {
+    async fn get_public_ip(&self) -> String {
+        let result = reqwest::get("http://whatismyip.akamai.com/")
```

Comment:
there must be a better way to do this lol, but any external calls should go into a top level file called clients

Reply:
Hehe it seems cheeky but theres really nothing wrong doing it this way. There are other alternatives though. I think we can query the dns or something, but this was simpler.

Reply:
But I moved it to the clients folder